actions/test/action.yml

```diff
@@ -0,0 +1,213 @@
+      env:
+        REQUIRED_DEPS: |
+          git
+          curl
+          gpg
+      run: |
+        apt-get update
+        for dep in $REQUIRED_DEPS; do
+          if ! dpkg -s "$dep" >/dev/null 2>&1; then
+            apt-get install -y "$dep"
+          fi
+        done
+
+    # Auto-detect coverage files if not specified
+    - id: detect-coverage-files
+      if: always() && inputs.coverage == 'github' && inputs.coverage-files == ''
+      uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+      env:
+        WORKING_DIRECTORY: ${{ inputs.working-directory }}
+      with:
+        script: |
+          const fs = require('node:fs');
+          const path = require('node:path');
+
+          const workingDirectory = process.env.WORKING_DIRECTORY || '.';
+          const workDir = path.resolve(process.env.GITHUB_WORKSPACE, workingDirectory);
+
+          // Common coverage file paths
+          const commonPaths = [
+            'coverage/cobertura-coverage.xml',
+            'coverage/coverage.xml',
+            'coverage/lcov.info',
+            'coverage/clover.xml',
+            'coverage/coverage-final.json',
+            'test-results/coverage.xml',
+            'test-results/cobertura-coverage.xml'
+          ];
+
+          for (const filePath of commonPaths) {
+            const fullPath = path.join(workDir, filePath);
+            if (fs.existsSync(fullPath)) {
+              core.info(`Auto-detected coverage file: ${filePath}`);
+              core.setOutput('coverage-files', filePath);
+              return;
+            }
+          }
+
+          core.warning('No coverage file auto-detected');
+          core.setOutput('coverage-files', 'coverage/cobertura-coverage.xml');
+
+    # Upload to Codecov
+    - name: 📊 Upload coverage to Codecov
+      if: always() && inputs.coverage == 'codecov'
+      uses: codecov/codecov-action@5a1091511ad55cbe89839c7260b706298ca349f7 # v5.5.1
+      with:
+        working-directory: ${{ inputs.working-directory }}
+        token: ${{ inputs.codecov-token }}
+        use_oidc: ${{ inputs.codecov-token == '' }}
+        disable_telem: true
+        fail_ci_if_error: false
+
+    # ReportGenerator for PR comments with coverage reports
+    - name: 📊 Generate coverage report
+      if: always() && inputs.coverage == 'github' && github.event_name == 'pull_request'
+      uses: danielpalme/ReportGenerator-GitHub-Action@d78d7a99ab2e53f4fa9e37c558feb9cd6a7a1c70 # v5.4.3
+      with:
+        reports: ${{ inputs.coverage-files != '' && inputs.coverage-files || steps.detect-coverage-files.outputs.coverage-files }}
+        targetdir: coveragereport
+        reporttypes: MarkdownSummaryGithub
+        sourcedirs: ${{ inputs.working-directory }}
+
+    - name: 📊 Add coverage PR comment
+      if: always() && inputs.coverage == 'github' && github.event_name == 'pull_request'
+      uses: marocchino/sticky-pull-request-comment@331f8f5b4215f0445d3c07b4967662a32a2d3e31 # v2.9.0
```

Comment:
Use this action instead: https://github.com/hoverkraft-tech/ci-github-common/tree/main/actions/create-or-update-comment